multiqc/modules/slamdunk/slamdunk.py

```diff
@@ -0,0 +1,832 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from Slamdunk """
+
+from __future__ import print_function
+import logging
+import re
+import numpy as np
+from distutils.version import StrictVersion
+from collections import OrderedDict
+
+from multiqc import config, BaseMultiqcModule, plots
+from curses.ascii import islower
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+    """
+    Slamdunk module class, parses slamdunk logs.
+    """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='Slamdunk', anchor='slamdunk',
+        href='https://github.com/t-neumann/slamdunk',
+        info="is a tool to analyze SLAMSeq data. "\
+         "More info to come.")
+
+        self.slamdunk_data = dict()
+        
+        self.PCA_data = dict()
+        
+        self.utrates_data = dict()
+        
+        self.rates_data_plus = dict()
+        self.rates_data_minus = dict()
+        
+        self.nontc_per_readpos_plus = dict()
+        self.nontc_per_readpos_minus = dict()
+        
+        self.tc_per_readpos_plus = dict()
+        self.tc_per_readpos_minus = dict()
+        
+        self.nontc_per_utrpos_plus = dict()
+        self.nontc_per_utrpos_minus = dict()
+        
+        self.tc_per_utrpos_plus = dict()
+        self.tc_per_utrpos_minus = dict()
+
+        # Switch string for plots to produce
+        produce = "" 
+
+        # Check whether summary also contains number of counted reads
+        extendedSummary = True
+            
+        for f in self.find_log_files(config.sp['slamdunk']['summary'], filehandles = True):
+            extendedSummary = extendedSummary and self.parseSummary(f)
+            
+        if len(self.slamdunk_data) == 0:
+            log.debug("Could not find any reports in {}".format(config.analysis_dir))
+        else :
+            produce += "summary"
+            
+        if extendedSummary:
+            log.info("Extended Slamdunk summary found.")
+            
+        log.info("Parsing PCA reports.")
+            
+        for f in self.find_log_files(config.sp['slamdunk']['PCA'], filehandles = True):
+            self.parsePCA(f)
+            
+        if len(self.PCA_data) == 0:
+            log.debug("Could not find PCA reports in {}".format(config.analysis_dir))
+        else :
+            produce += "PCA"
+            
+        log.info("Parsing UTR rates reports. This can take a while...")
```

Comment:
Yeah it's possible. Right now I'm producing the boxplots we discussed using this file and I would really dislike producing another extra file with the same data in another format just for MultiQC. But now that I think of it, I could produce the median values and add it to the header - maybe that's the best way to go